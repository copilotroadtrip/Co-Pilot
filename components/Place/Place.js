import React, { Component } from "react";
import {
  StyleSheet,
  Text,
  View,
  TextInput,
  TouchableOpacity,
  ImageBackground
} from "react-native";
import { cityImages } from "../../assets/cityImages";

export default class Place extends Component {
  constructor(props) {
    super(props);
    this.state = { randomCity: "" };
  }

  componentDidMount = () => {
    this.randomCityGenerator();
  };

  randomCityGenerator = () => {
    const { navigation } = this.props;
    const place = navigation.getParam("placeData");
    const randomIndex = Math.floor(Math.random() * 3);
    if (place.population > 500000) {
      this.setState({ randomCity: cityImages.large[randomIndex] });
    }
    if (place.population < 500000) {
      this.setState({ randomCity: cityImages.medium[randomIndex] });
    }
    if (place.population < 2000) {
      this.setState({ andomCity: cityImages.small[randomIndex] });
    }
  };

  render() {
    return (
      <ImageBackground
        style={styles.backGround}
        source={{ uri: this.state.randomCity }}
      >
        <Text>stuff</Text>
      </ImageBackground>
    );
  }
}

const styles = StyleSheet.create({
  backGround: {
    width: "100%",
    height: "100%"
  }
});
