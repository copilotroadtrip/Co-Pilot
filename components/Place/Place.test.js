import React from "react";
import NavigationTestUtils from "react-navigation/NavigationTestUtils";
import { mockTrip } from "../../utilities/mockTrips.js";
import { shallow } from "enzyme";

import Place from "./Place";


describe("Place", () => {
  NavigationTestUtils.resetInternalState();
  let wrapper;
  const createTestProps = props => ({
    navigation: {
      getParam: jest.fn(() => {
        return mockTrip.places[0];
      })
    },
    ...props
  });

  beforeEach(() => {
   
    props = createTestProps({});
    wrapper = shallow(<Place {...props} />);
  });

  it(`matches snapshot`, () => {
    expect(wrapper).toMatchSnapshot();
  });
});
