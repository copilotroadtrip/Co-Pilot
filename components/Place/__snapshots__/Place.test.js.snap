// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Place matches snapshot 1`] = `ShallowWrapper {}`;
